auth/core/multiauth.go

```diff
@@ -50,7 +47,7 @@ func (m *MultiAuth) Authenticate(req *http.Request) (string, error) {
 
 func (m *MultiAuth) Verify(data AuthRequest, expected string) error {
 	var strs []string
-	err := json.Unmarshal([]byte(data.Data), &strs)
+	err := json.Unmarshal([]byte(expected), &strs)
```

Comment:
we were unmarshaling `data.Data` instead of `expected`, which is the string generated by the `Authenticate` call